src/quoridor/game/fence.go

```diff
@@ -10,6 +10,46 @@ func (f Fence) Equals(other Fence) bool {
 	return f.NWSquare.Equals(other.NWSquare) && f.Horizontal == other.Horizontal
 }
 
+type Fences []Fence
+
+func (fences Fences) indexOf(e Fence) int {
+	for index, a := range fences {
+		if a.Equals(e) {
+			return index
+		}
+	 }
+	return -1
+}
+
+func Collision(fences Fences, fence1 Fence, fence2 Fence) bool {
```

Comment:
Why did you test on 2 fences?

Reply:
because in any direction you can have two fences which can block the path